Cache/Memoize.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Memoize
+ * @param {Function} fn
+ * @returns
+ */
+export const memoize = (func) => {
+  // eslint-disable-next-line no-console
+  console.log(`Creating cache for function '${func.name}'`)
```

Comment:
Those `console.log()` are probably not crucial for the functionality, and don't look good when running tests.

Reply:
While on the test subject. Where did you get the idea to name the directory `__test__`, when the rest is just `test` ?

Reply:
Oh! You're right. I apply my own convention rather than yours... My apologize 🙏 
I'm gonna fix it!